File: 1.two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dict = {}
        for i, val in enumerate(nums):
            if val in num_dict:
                return [num_dict[val], i]

            else:
                num_dict[target-val] = i
            
        

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        set<int> s;
        map<int, int> m;
        vector<int> v;
        for (int i=0;i<nums.size();i++) {
            if (s.count(nums[i])) {
            v.push_back(i); v.push_back(m[target-nums[i]]); 
            break;
}
            m[nums[i]] = i;
            s.insert(target-nums[i]);
        }
        return v;
        
    }
};